File: 07_Polymorphism/Basics/Main.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    virtual void sound() {
        cout << "This is an animal sound." << endl;
    }
};

class Dog : public Animal {
public:
    void sound() override {
        cout << "The dog barks." << endl;
    }
};

int main() {
    Animal* animalPtr;
    Dog dog;

    animalPtr = &dog;

    animalPtr->sound(); 

    return 0;
}

File: 07_Polymorphism/Basics/Main.java
class Animal {
    void sound() {
        System.out.println("This is an animal sound.");
    }
}

class Dog extends Animal {
    void sound() {
        System.out.println("The dog barks.");
    }
}

public class Main {
    public static void main(String[] args) {
        Animal animalRef;
        Dog dog = new Dog();

        animalRef = dog;

        animalRef.sound(); 
    }
}
